library/core/src/char/methods.rs

```diff
@@ -1783,14 +1785,11 @@ pub fn encode_utf8_raw(code: u32, dst: &mut [u8]) -> &mut [u8] {
             *c = (code >> 6 & 0x3F) as u8 | TAG_CONT;
             *d = (code & 0x3F) as u8 | TAG_CONT;
         }
-        _ => panic!(
-            "encode_utf8: need {} bytes to encode U+{:X}, but the buffer has {}",
-            len,
-            code,
-            dst.len(),
-        ),
+        // Note that the original message is not const-compatible due to formatting.
```

Comment:
'original message' won't make sense here once this is merged - maybe best to just have this as a PR comment